packages/protocol/contracts/shared/tokenvault/ERC20Solver.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../libs/LibStrings.sol";
+import "../../layer1/based/ITaikoInbox.sol";
+import "../common/EssentialContract.sol";
+
+/// @title ERC20Solver
+/// @notice This enables bridging intent solving feature in erc20vault and facilitates
+/// fast withdrawals of tokens when bridging from l2 to l1.
+/// @custom:security-contact security@taiko.xyz
+contract ERC20Solver is EssentialContract {
+    /// @dev Represents an operation to solve an ERC20 bridging intent on destination chain
+    struct SolveOp {
```

Comment:
In AA, there is a new term called UserOperation, here, shall we call it "SolverOp" instead of "SolveOp"?